### PR TITLE
feat: zstd-compress requests over 1MB

### PR DESCRIPTION
**Pull request by jmoralez**
Automatically compresses payloads over 1MB using [zstandard](https://github.com/facebook/zstd). This should help reduce the time when sending large payloads over slow internet connections.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -48,6 +48,7 @@
         "tenacity",
         "tqdm",
         "utilsforecast>=0.2.8",
+        "zstandard",
```
Comment (elephaint) [inline setup.py:51]: Should we also set `httpx[zstd]` to ensure correct version is installed?

Comment (jmoralez) [reply setup.py]: Oh it seems httpx automatically adds `accept-encoding: 'zstd'` to the headers if zstandard is installed and modal automatically compresses the responses when it sees this, so it's indeed worth adding. Thanks!

File: setup.py [hunk 2]
```diff
@@ -42,8 +42,7 @@
     python_requires=">=3.9",
     install_requires=[
         "annotated-types",
-        "coreforecast>=0.0.14",
```
Comment (jmoralez) [inline setup.py:45]: This was added in a bad merge in #547

File: nbs/src/nixtla_client.ipynb [hunk 3]
```diff
@@ -2249,6 +2267,48 @@
...
+    "def capture_request():\n",
+    "    original_client = httpx.Client\n",
+    "    httpx.Client = CapturingClient\n",
+    "    try:\n",
+    "        yield\n",
+    "    finally:\n",
+    "        httpx.Client = original_client\n",
+    "\n",
+    "# this produces a 1MB payload\n",
+    "series = generate_series(250, n_static_features=2)\n",
```
Comment (elephaint) [inline nbs/src/nixtla_client.ipynb:2302]: I thought it only compresses above 1 mb? So a 1 mb payload shouldn't compress?

**Review by elephaint — APPROVED** [review]
LGTM, two insignificant comments:

- Test is with 1 MB but I thought the idea that it doesn't compress up to (and including) 1 MB?
- Out of curiosity why 1MB as threshold?

**Comment by jmoralez** [comment]
I meant over 1MB in the comment, it's just slightly above it.

The threshold is because at 1MB it takes around 1ms to compress and it's most likely worth it due to the savings in bandwidth, since it always compresses to less than half the original size.

**Pull request merged**